### PR TITLE
Controller initialize

### PR DESCRIPTION
**Pull request by joshjung**

### CODE REVIEW

File: src/Bus.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    let p = this.parent;
+
+    while (p) {
+      p._dispatch(event);
+      p = p.parent;
+    }
+  }
+
+  /**
+   * Internal dispatch tha
```
Comment (corragon) [inline src/Bus.js]: Incomplete comment

File: src/Controller.js [hunk 2]
```diff
@@ -17,17 +17,21 @@ class Controller extends RingaObject {
...
     super(id);
 
-    if (__DEV__ && !domNode) {
-      throw Error('Controller:constructor(): no DOMNode provided to constructor!');
+    if (bus && !(typeof bus.addEventListener === 'function')) {
+      if (__DEV__ && options) {
+        throw Error('Controller(): invalid bus or DOM node passed into constructor!');
+      }
+
+      options = bus;
```
Comment (corragon) [inline src/Controller.js]: I don't think lines 26-31 are correct.  It won't throw an error if bus is undefined, but it did if __DEV__ were true and domNode was undefined.

File: src/Controller.js [hunk 3]
```diff
@@ -52,10 +56,58 @@ class Controller extends RingaObject {
...
+  busMounted(bus) {
+    // To be overridden
+  }
+
+  /**
+   * Takes in a series of event types and converts them to static properties on the parent class.
+   *
+   * For example:
+   *
+   *   controler = new MyController();
```
Comment (corragon) [inline src/Controller.js]: Typo

File: src/Controller.js [hunk 4]
```diff
@@ -67,15 +119,28 @@ class Controller extends RingaObject {
...
 
-  getListener(eventType) {
+  /**
+   * Returns true if there is a listener for the provided eventType.
+   *
+   * @param eventType The event type - a String.
+   * @returns {*} True if the event type has a listener.
+   */
+  getThreadFactoryFor(eventType) {
     return this.eventTypeToThreadFactory[eventType];
```
Comment (corragon) [inline src/Controller.js]: Should this line be:
`return !!this.eventTypeToThreadFactory[eventType];`

File: src/RingaEvent.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
 import RingaObject from './RingaObject';
 import ErrorStackParser from 'error-stack-parser';
 import {ringaEventToDebugString} from './util/debug';
+import {isDOMNode} from './util/type';
 
 let eventIx = 0;
 
 /**
- * RingaEvent wraps a CustomEvent that is dispatched on the DOM:
+ * RingaEvent is a generic event type for Ringa that, when dispatched on the DOM, wraps a CustomEvent:
```
Comment (corragon) [inline src/RingaEvent.js]: DOM or bus

File: src/Controller.js (hunk 4 above)
Comment (joshjung) [inline src/Controller.js]: The comment is wrong, will update.

**Pull request merged**